sympy/core/tests/test_constructor_postprocessor.py

```diff
@@ -0,0 +1,45 @@
+from sympy import Symbol, Mul, symbols
+
+
+class SymbolInMulOnce(Symbol):
+    # Test class for a symbol that can only appear once in a `Mul` expression.
+
+    _constructor_postprocessor_dict = {
+        "Mul": [lambda x: x],
+        "Pow": [lambda x: x.base if isinstance(x.base, SymbolInMulOnce) else x],
+        "Add": [lambda x: x],
+    }
+
+
+class SymbolRemovesOtherSymbols(Symbol):
+    # Test class for a symbol that removes other symbols in `Mul`.
+
+    def postprocess(expr):
```

Comment:
It's confusing to not have it be static, since otherwise `expr` would really be `self` on an instance.